src/routes/leaderboard.ts

```diff
@@ -1,45 +1,42 @@
 import express, { Router } from "express";
 import { Client } from "pg";
+import { useErrorHandler } from "../utils/errorHandler";
+import { z } from "zod";
 
 interface LeaderboardRow {
     id: number;
     breed: string;
     votes: number;
 }
 
-export function createLeaderboardRouter(client: Client): Router {
+export default function createLeaderboardRouter(client: Client): Router {
     const router = express.Router();
 
-    router.get<{}, LeaderboardRow[] | string>("/", async (_req, res) => {
-        try {
+    router.get<{}, LeaderboardRow[]>(
+        "/",
+        useErrorHandler(async (_req, res) => {
             const result = await client.query(
                 "SELECT * FROM leaderboard ORDER BY votes DESC LIMIT 10"
             );
             res.status(200).json(result.rows);
-        } catch (error) {
-            console.error(error);
-            res.status(500).send("An error occurred. Check server logs.");
-        }
-    });
+        })
+    );
 
-    router.put<{}, LeaderboardRow | string, { breed: string }>(
+    router.put<{}, LeaderboardRow, Pick<LeaderboardRow, "breed">>(
         "/",
-        async (req, res) => {
-            try {
-                const result = await client.query(
-                    `INSERT INTO leaderboard (breed)
+        useErrorHandler(async (req, res) => {
+            const mySchema = z.string().min(1);
```

Comment:
Schema should be defined in global scope, we don't want to create the object every time the route is accessed. This way we can also infer the types later `z.infer<...>`.